### ISSUE TITLE
[StardogConnecter] Reasoning (on/off) by query

### ISSUE THREAD

**Issue by charbull** [dotnetrdf/dotnetrdf#108]
Hello, 

Stardog supports reasoning by query to be activated on/off, 
When it is activated, then the reasoning applied will be the one in the configuration.

In the [Java API](https://www.stardog.com/docs/java/snarl/com/complexible/stardog/api/query) it is possible to activate the reasoning by query:
```java
Query<T>	reasoning(boolean theReasoningFlag)
Specify whether you would like the query to be executed with reasoning.
```

Is there something similar in the StardogConnector?
In terms of implementation, if StarodgConnector relies on http, this reasoning is activated by appending "reasoning=true" to the `Http get`.

I submitted the pull request [#109](https://github.com/dotnetrdf/dotnetrdf/pull/109)

**Comment by kal** [comment]
There is more extensive reasoning support than that already in the connector. You can set the reasoning mode on the connector using the Reasoning property. It allows you to not only turn the reasoning on and off but also to specify the type of reasoning you want applied. So I think that this is already covered.

**Comment by charbull** [comment]
@kal in Stardog there is a difference between:
1. Reasoning mode at the database level
2. Reasoning activated/dis-activated by query, when the reasoning is activated by query it will apply the reasoning mode selected at the database level.

I took some screenshots to illustrate how Stardog operates.

## Reasoning Mode is configured at the Database level:
![reasoningmode](https://user-images.githubusercontent.com/1880238/27049481-80aeac96-4f7d-11e7-9467-04a02f2fa5ba.png)

## Reasoning by Query, activated or not for each query in the query panel:
![reasoningbyquery](https://user-images.githubusercontent.com/1880238/27049485-834b1606-4f7d-11e7-815a-3ec38eafc524.png)
When the reasoning is activated, the REST Get call to the Stardog database appends `reasoning=true` which is then handled by the database. 

You can try StardogConnector and the modified code in the pull request, you will notice the difference in the results

The StardogConnector supports only the reasoning mode and **not the reasoning by query**.

In the [Java API](https://www.stardog.com/docs/java/snarl/com/complexible/stardog/api/query) it is possible to activate the reasoning by query:
```java
Query<T>	reasoning(boolean theReasoningFlag)
Specify whether you would like the query to be executed with reasoning.
```

Hope that helps,
Charbel

**Comment by kal** [comment]
Thanks @charbull that makes sense. I've a couple of comments on the PR, but I'll add them there instead of here to make it easier to follow.

**Comment by kal** [comment]
Fixed in PRs #113 and #115
